### PR TITLE
Static Code Analysis warnings fixed (partially)

### PR DESCRIPTION
**Pull request by kalessil**
Code cleanup (actually code base is in very good state): code style, if conditionals, strict types, performance. Tests are green.

Following tools are used for code analysis: PhpStorm + <a href="https://plugins.jetbrains.com/plugin/7622?pr=phpStorm">Php Inspections (EA Extended)</a>.

### CODE REVIEW

File: src/Facebook/Entities/AccessToken.php [hunk 1]
```diff
@@ -178,7 +177,7 @@ public static function getAccessTokenFromCode($code, $appId = null, $appSecret =
   {
     $params = array(
       'code' => $code,
-      'redirect_uri' => '',
```
Comment (SammyK) [inline src/Facebook/Entities/AccessToken.php]: I'm :-1: for removing trailing commas from arrays. It doesn't cause any errors and is super convenient when adding new items. One of the nice features of PHP that JavaScript got wrong. :)

File: src/Facebook/FacebookRequest.php [hunk 2]
```diff
@@ -194,13 +194,13 @@ public function __construct(
...
-      && !isset($params["access_token"])) {
-      $params["access_token"] = $session->getToken();
+      && !isset($params['access_token'])) {
+      $params['access_token'] = $session->getToken();
     }
-    if (FacebookSession::useAppSecretProof()
-      && !isset($params["appsecret_proof"])) {
-      $params["appsecret_proof"] = $this->getAppSecretProof(
-        $params["access_token"]
+    if (!isset($params['appsecret_proof'])
```
Comment (SammyK) [inline src/Facebook/FacebookRequest.php]: Space between `!` and `isset` so it's more obvious it's negated. And it's still PSR-2 compliant.

File: src/Facebook/FacebookRequest.php [hunk 3]
```diff
@@ -249,7 +249,7 @@ public function execute()
 
     static::$requestCount++;
 
-    $etagHit = 304 == $connection->getResponseHttpStatusCode();
+    $etagHit = (304 === $connection->getResponseHttpStatusCode());
```
Comment (SammyK) [inline src/Facebook/FacebookRequest.php]: The parentheses around this statement is redundant and harder to read so it should be like it was before. The strict `===` checking is OK though.

File: src/Facebook/FacebookRequestException.php [hunk 4]
```diff
@@ -80,7 +80,7 @@ public static function create($raw, $data, $statusCode)
     if (!isset($data['error']['code']) && isset($data['code'])) {
       $data = array('error' => $data);
     }
-    $code = (isset($data['error']['code']) ? $data['error']['code'] : null);
+    $code = (isset($data['error']['code']) ? (int)$data['error']['code'] : null);
```
Comment (SammyK) [inline src/Facebook/FacebookRequestException.php]: Space between `(int)` and `$data` to be consistent with the rest of the code in the SDK.

File: src/Facebook/HttpClients/FacebookCurl.php [hunk 5]
```diff
@@ -33,9 +33,9 @@ class FacebookCurl
 {
 
   /**
-   * @var resource Curl resource instance
+   * @var resource|null Curl resource instance
    */
-  protected $curl = null;
```
Comment (SammyK) [inline src/Facebook/HttpClients/FacebookCurl.php]: There was a recent PR that added this. You'll need to duke it out with the original PR. :) See: #328 & #327.

**Comment by SammyK** [comment]
:+1: except for the things I commented on. :)

File: src/Facebook/Entities/AccessToken.php (hunk 1 above)
Comment (kalessil) [inline src/Facebook/Entities/AccessToken.php]: And it's also a open door for copy paste =) I will revert this changes nevertheless.

File: src/Facebook/FacebookRequest.php (hunk 2 above)
Comment (kalessil) [inline src/Facebook/FacebookRequest.php]: Right, has sense.

File: src/Facebook/FacebookRequest.php (hunk 3 above)
Comment (kalessil) [inline src/Facebook/FacebookRequest.php]: Will remove, for me it's easier to read with parentheses.

File: src/Facebook/FacebookRequestException.php (hunk 4 above)
Comment (kalessil) [inline src/Facebook/FacebookRequestException.php]: Right

File: src/Facebook/HttpClients/FacebookCurl.php (hunk 5 above)
Comment (kalessil) [inline src/Facebook/HttpClients/FacebookCurl.php]: That something interesting, thanks for links.

**Comment by kalessil** [comment]
@SammyK : reverted commas, null assignments, added spaces for casting and !isset.

**Comment by SammyK** [comment]
:+1:

**Pull request merged**